src/components/Search/SearchByNameResults.tsx

```diff
@@ -0,0 +1,48 @@
+import { useTranslation } from 'react-i18next'
+import { UDTQueryResult } from '../../services/ExplorerService/fetcher'
+import styles from './SearchByNameResults.module.scss'
+import EllipsisMiddle from '../EllipsisMiddle'
+
+type Props = {
+  udtQueryResults: UDTQueryResult[]
+  truncateTypeHash?: boolean
+}
+
+export const SearchByNameResults = (props: Props) => {
+  const { udtQueryResults, truncateTypeHash } = props
+  return (
+    <div className={styles.searchResultsPanelWrapper}>
+      {udtQueryResults.length === 0 ? (
+        <EmptySearchByNameResult />
+      ) : (
+        udtQueryResults.map(item => {
+          return <SearchByNameResult key={item.typeHash} item={item} truncateTypeHash={truncateTypeHash} />
+        })
+      )}
+    </div>
+  )
+}
+
+const EmptySearchByNameResult = () => {
+  const { t } = useTranslation()
+  return <>{t('search.no_search_result')}</>
+}
+
+const SearchByNameResult = (props: { item: UDTQueryResult; truncateTypeHash?: boolean }) => {
+  const { t } = useTranslation()
+  const { item, truncateTypeHash } = props
+  const { typeHash, symbol } = item
+  return (
+    <>
+      <a className={styles.searchResult} href={`${window.origin}/sudt/${typeHash}`}>
+        <span className={styles.tokenSymbol}>{symbol ?? t('udt.unknown_token')}</span>
+        {truncateTypeHash ? (
```

Comment:
I think this should be unconditional. There doesn't seem to be any reason in this interface to not use ellipsis when the width is insufficient.

Reply:
under some condition the page size is enought to show full hash

![image](https://github.com/Magickbase/ckb-explorer-frontend/assets/3870972/c39975b7-e747-47f4-a01e-67301f3c35f7)

Reply:
`EllipsisMiddle` will display the entire content when the width is sufficient, so it can be removed.

The design here is actually quite strange. Even if `EllipsisMiddle` is not used, exposing `truncateTypeHash` directly to external control is not appropriate. It should provide more granular control, such as `maxDisplayedHashLength`, or allow the external code to pass in the truncated hash, or provide a `getDisplayedHash: (item: UDTQueryResult) => string` function.

When designing interfaces (`props`) between components, it is beneficial to think more about the relationship between them and then decide on the appropriate level of abstraction. This can help improve your code design.

Reply:
replaced with `EllipsisMiddle`